libraries/scopes/src/simplify.js

```diff
@@ -0,0 +1,58 @@
+const { normalizeScopeSet, scopeCompare } = require("./normalize");
+
+exports.simplifyScopeExpression = scopeExpression => {
+  let key = Object.keys(scopeExpression);
+  if (Object.values(scopeExpression)[0].length === 1) {
+    if (key.includes("AnyOf")) {
+      return scopeExpression.AnyOf[0];
+    } else if (key.includes("AllOf")) {
+      return scopeExpression.AllOf[0];
+    }
+  }
```

Comment:
I think this section handles the case where AnyOf and AllOf have a single option.  But what if they have multiple options that *simplify* to a single option?  This is a place where fixed-point iteration would help -- perhaps a previous round converted `{AnyOf: ["scope1", "scope2", "scope*"]}` to `{AnyOf: ["scope*"]}`, and this round then simplifies that further to `"scope*"`.

Reply:
Yes. I understand now.